project-keeper/src/main/java/com/exasol/projectkeeper/sources/analyze/golang/GolangDependencyChangeCalculator.java

```diff
@@ -57,17 +59,37 @@ private List<DependencyChange> getDependencyChanges(final Type type) {
     }
 
     private Type getType(final DependencyChange change) {
-        final String name = change.getArtifactId();
-        if (isGolangRuntime(name)) {
+        final String moduleName = change.getArtifactId();
+        if (isGolangRuntime(moduleName)) {
             return Type.COMPILE;
         }
         return this.dependencies.getDependencies().stream() //
-                .filter(dep -> dep.getName().equals(name)) //
+                .filter(dep -> dep.getName().equals(moduleName)) //
                 .map(ProjectDependency::getType) //
                 .findFirst() //
-                .orElseThrow(() -> new IllegalStateException(ExaError.messageBuilder("E-PK-CORE-148").message(
-                        "Error finding type of dependency {{artifactId}}, all available dependencies: {{all dependencies}}.",
-                        name, this.dependencies.getDependencies()).ticketMitigation().toString()));
+                .orElseGet(() -> getTypeByPrefix(moduleName));
+    }
+
+    private Type getTypeByPrefix(final String moduleName) {
+        if (moduleName.matches(".*/v\\d+")) {
+            final String moduleNamePrefix = moduleName.substring(0, moduleName.lastIndexOf("/"));
+            LOGGER.finest(() -> "Found prefix '" + moduleNamePrefix + "' for module '" + moduleName + "'.");
+            final Optional<Type> type = this.dependencies.getDependencies().stream() //
+                    .filter(dep -> dep.getName().startsWith(moduleNamePrefix)) //
```

Comment:
is `startsWith()`  really safe?

Reply:
What do you mean with safe? It's a simple solution that works.